libflux/src/flux/benches/basic.rs

```diff
@@ -0,0 +1,24 @@
+#[macro_use]
+extern crate criterion;
+extern crate flux;
+
+use criterion::Criterion;
+use flux::parser::Parser;
+
+// everything.flux was generated by running the following
+// command from this directory. This file is trimmed to 5000
+// lines, as the 52k lines pre-trim would cause the benchmark
+// test to take a very long time (almost three hours).
+// find ../../../.. -name "*.flux" | cat >> everything.flux && sed -i '' '5001,$ d' everything.flux
+fn everything(c: &mut Criterion) {
+    let flux = include_str!("./everything.flux");
+    c.bench_function("everything.flux", |b| {
+        b.iter(|| {
+            let mut parser = Parser::new(flux);
+            parser.parse_file("".to_string());
```

Comment:
Does rust optimize away this code if the return value is never used? I know Go has some optimizations that you have to be careful to avoid in these simple benchmark tests. I.e. set the return value to a package level var etc.

Reply:
My understanding was that the code wasn't optimized at all. I checked, and the result isn't optimized away, but I added an explicit call to `black_box` which makes the function opaque to the optimizer (and thus isn't optimized). As a result, sample gathering is faster (<5 mins for 100 samples) and the result is much more stable.